edx_proctoring/views.py

```diff
@@ -366,10 +366,18 @@ def get(self, request):
         details = learning_sequences_service.get_user_course_outline_details(
             course_key, user, pytz.utc.localize(datetime.max)
         )
-
+        due_date = details.schedule.course_end or pytz.utc.localize(datetime.max)
         for onboarding_exam in onboarding_exams:
             usage_key = BlockUsageLocator.from_string(onboarding_exam.content_id)
-            if usage_key not in details.outline.accessible_sequences:
+            exam_schedule_item = details.schedule.sequences.get(usage_key)
+            if exam_schedule_item and exam_schedule_item.due:
+                due_date = exam_schedule_item.due
+
+            user_outline_detail = learning_sequences_service.get_user_course_outline_details(
```

Comment:
@ormsbee and @Zacharis278 I am calling the `get_user_course_outline_details` a second time here. This is because I needed the scheduledItem from the first call. Please advice if you have a better way to get course_end_date or exam due dates

Reply:
I think it's fine to do this. Minor note that this second call, you probably don't need the `get_user_course_outline_details`, since you're only looking at whether the sequence is accessible or not and `get_user_course_outline` is sufficient to give you that.

Reply:
In theory, you could get those specific dates more quickly from edx-when's API directly, but in practice, edx-when always pre-loads scheduling for the whole course anyway, so you don't actually save any time.

Reply:
@ormsbee Took your advice on using `get_user_course_outline` for the second call.